hpx/runtime/threads/detail/scheduling_loop.hpp

```diff
@@ -420,6 +422,23 @@ namespace hpx { namespace threads { namespace detail
                             num_thread);
                         scheduler.SchedulingPolicy::do_some_work(num_thread);
                     }
+                    else if (HPX_UNLIKELY(state_val == pending_boost))
+                    {
+                        if (HPX_LIKELY(next_thrd == nullptr)) {
+                            // schedule other work
+                            scheduler.SchedulingPolicy::wait_or_add_new(
+                                num_thread, this_state.load() < state_stopping,
+                                idle_loop_count);
+                        }
+
+                        thrd->set_state(pending);
+
+                        // schedule this thread again immediately with boosted
+                        // priority
+                        scheduler.SchedulingPolicy::schedule_thread(thrd,
+                            num_thread, thread_priority_boost);
```

Comment:
`pending_boost` is used in `hpx::lcos::local::spinlock` right now. When putting the thread to the *end* of the queue, it might get stolen. I'd argue that in the case of a spinlock, one really wants to avoid getting stolen, also reducing the overheads of pushing/popping to the queue might be favorably here. Since spinlocks should be used in short critical sections only anyway, this would lead to faster turnaround times. You could argue that in cases where this busy loop takes too long, a proper mutex might be more appropriate.